dask/dataframe/shuffle.py

```diff
@@ -371,6 +371,7 @@ def shuffle(
     shuffle_disk
     """
     list_like = pd.api.types.is_list_like(index) and not is_dask_collection(index)
+    shuffle = shuffle or get_default_shuffle_algorithm()
```

Comment:
Again - If we will ultimately use a task-based shuffle, we want to avoid the memory overhead of creating/assigning a new column by going into the `if`-block below. We cannot do this if `shuffle` is still `None`.